tests/test_optimizer.py

```diff
@@ -694,3 +696,118 @@ def test_query_nested_fragments(db, gql_client: GraphQLTestClient):
         res = gql_client.query(query)
 
     assert res.data == expected
+
+
+@pytest.mark.django_db(transaction=True)
+def test_query_annotate(db, gql_client: GraphQLTestClient):
+    query = """
+      query TestQuery ($node_id: GlobalID!) {
+        project (id: $node_id) {
+          id
+          name
+          isDelayed
+          milestonesCount
+        }
+      }
+    """
+
+    expected = []
+    today = timezone.now().date()
+    for p in ProjectFactory.create_batch(2):
+        ms = MilestoneFactory.create_batch(3, project=p)
+        p_res: dict[str, Any] = {
+            "id": to_base64("ProjectType", p.id),
+            "name": p.name,
+            "isDelayed": p.due_date < today,
+            "milestonesCount": len(ms),
+        }
+        expected.append(p_res)
+    for p in ProjectFactory.create_batch(
+        2,
+        due_date=today - datetime.timedelta(days=1),
+    ):
+        ms = MilestoneFactory.create_batch(2, project=p)
+        p_res: dict[str, Any] = {
+            "id": to_base64("ProjectType", p.id),
+            "name": p.name,
+            "isDelayed": p.due_date < today,
+            "milestonesCount": len(ms),
+        }
+        expected.append(p_res)
+
+    assert len(expected) == 4
+    for e in expected:
+        if DjangoOptimizerExtension.enabled.get():
+            with assert_num_queries(1):
+                res = gql_client.query(query, {"node_id": e["id"]})
+                assert res.data == {"project": e}
+        else:
+            # isDelayed and milestonesCount requires the optimizer to be turned on
+            res = gql_client.query(
+                query,
+                {"node_id": e["id"]},
+                asserts_errors=False,
+            )
+            assert res.errors
+
+
+@pytest.mark.django_db(transaction=True)
+def test_query_annotate_with_callable(db, gql_client: GraphQLTestClient):
```

Comment:
I'm not sure if this test is testing the case of prefetch + annotate, i.e., the annotate is called inside the custom queryset of a `prefetch_related`. This is something we should test.